client/src/pages/SignUp.js

```diff
@@ -0,0 +1,250 @@
+import { makeStyles } from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+import Background from '../elements/Background';
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(20),
+    padding: theme.spacing(5),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center'
+  },
+  headerText: {
+    marginTop: theme.spacing(10)
+  },
+  form: {
+    width: '100%',
+    marginTop: theme.spacing(3),
+    textAlign: 'center'
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+    padding: theme.spacing(1, 6),
+    border: 0,
+    borderRadius: 25,
+    color: '#fff',
+    textTransform: 'none',
+    fontSize: 15
+  }
+}));
+
+export default function SignIn() {
+  const classes = useStyles();
+
+  // ========== FORM VALIDATION LOGIC BEGIN ==========
+
+  // Validate username
+  const [ username, setUsername ] = useState({
+    value: '',
+    error: 'Field can not be blank',
+    touched: false
+  });
+
+  const usernameHandler = (event) => {
+    const usernameRegex = /^[a-zA-Z0-9-]+$/;
+    let errorFound = '';
+    if (event.target.value.length < 4 || event.target.value.length > 16) {
+      errorFound = 'Username length should be 4-16 chars long';
+    } else if (!event.target.value.match(usernameRegex)) {
+      errorFound = 'Only letters, numbers and  "-" are  acceptable';
+    }
+    setUsername({
+      value: event.target.value,
+      touched: true,
+      error: errorFound
+    });
+  };
+
+  // Validate email
+  const [ email, setEmail ] = useState({
+    value: '',
+    error: 'Field can not be blank',
+    touched: false
+  });
+
+  const emailHandler = (event) => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    let errorFound = '';
+    if (!event.target.value.match(emailRegex)) {
+      errorFound = 'Incorrect email address';
+    }
+    setEmail({
+      value: event.target.value,
+      touched: true,
+      error: errorFound
+    });
+  };
+
+  //Validate password
+  const [ password, setPassword ] = useState({
+    value: '',
+    error: 'Field can not be blank',
+    touched: false
+  });
+
+  const passwordHandler = (event) => {
+    let errorFound = null;
+    if (event.target.value.length < 6) {
+      errorFound = 'Password must be at least 6 characters long';
+    } else if (!event.target.value.match(/(?=.*[a-z])/)) {
+      errorFound = 'Password must contain at least one lower case letter';
+    } else if (!event.target.value.match(/(?=.*\d)/)) {
+      errorFound = 'Password must contain at least one digit';
+    } else if (!event.target.value.match(/(?=.*[A-Z])/)) {
+      errorFound = 'Password must contain at least one upper case letter';
+    } else if (!event.target.value.match(/(?=.*[!@#$%^&*])/)) {
+      errorFound = 'Password must contain at least one special char';
+    } else if (repeatedPassword.value && event.target.value !== repeatedPassword.value) {
+      errorFound = 'Passwords do not match';
+    }
```

Comment:
Good case for a switch statement.

Reply:
Thougth of it, but with  'break;' it's gonna be the same number of lines. But maybe more readable. Thx for suggestion

Reply:
Readability is part of it, a run of `else if` is also the least performant way of dealing with multiple cases in JS. Not really an issue with something this small, but good practice to avoid.